src/Components/Web.JS/src/Boot.WebAssembly.ts

```diff
@@ -116,7 +119,54 @@ async function boot(options?: Partial<WebAssemblyStartOptions>): Promise<void> {
     }
   };
 
+  type BeforeBlazorStartedCallback = (options: Partial<WebAssemblyStartOptions> | undefined, bootConfigResul: BootConfigResult) => Promise<void>;
+  type AfterBlazorStartedCallback = () => Promise<void>;
+  type BlazorInitializer = { beforeBlazorStarts: BeforeBlazorStartedCallback, afterBlazorStarted: AfterBlazorStartedCallback };
+
   const bootConfigResult = await bootConfigPromise;
+  const bootConfig = bootConfigResult.bootConfig;
+  const libraryInitializersTimeout = options?.libraryInitializersTimeout ?? 30;
+  const afterBlazorStartedCallbacks: AfterBlazorStartedCallback[] = [];
+  if (bootConfigResult.bootConfig.libraryInitializers) {
+    const initializerFiles = bootConfigResult.bootConfig.libraryInitializers;
+    try {
+      await Promise.all(Object.entries(initializerFiles).map(f => importAndInvokeInitializer(...f)));
+    } catch (error) {
+      console.warn(`A library initializer produced an error: '${error}'`);
+    }
+  }
+
+  function adjustPath(path: string): string {
+    // This is the same we do in JS interop with the import callback
+    const base = document.baseURI;
+    path = base.endsWith('/') ? `${base}${path}` : `${base}/${path}`;
+    return path;
+  }
+
+  async function importAndInvokeInitializer(path: string, signature: string): Promise<void> {
+    const adjustedPath = adjustPath(path);
+    const initializer = await import(/* webpackIgnore: true */ adjustedPath) as Partial<BlazorInitializer>;
+    if (initializer === undefined) {
+      return;
+    }
+    const { beforeBlazorStarts, afterBlazorStarted } = initializer;
+    if (afterBlazorStarted) {
+      afterBlazorStartedCallbacks.push(afterBlazorStarted);
+    }
+    if (beforeBlazorStarts) {
+      const timeout = libraryInitializersTimeout * 1000;
+      await Promise.race([
+        beforeBlazorStarts(candidateOptions, bootConfigResult),
+        // By default we give each library 30 seconds to load and do whatever they have to do, before we continue loading.
+        new Promise<void>((resolve) =>
+          setTimeout(() => {
+            console.log(`Library '${path}' took longer than '${libraryInitializersTimeout}' to initialize.`);
+            resolve();
+          },
+            timeout))]);
+    }
+  }
```

Comment:
Can we move all this new logic out into a separate file? The Boot.*.ts files are already a big collection of semi-related stuff so it would be great to minimize the amount of new logic that shows up all in one place.

Since an equivalent is needed in Server/WebView too, this would also help to avoid duplicating definitions for all the types.

Reply:
done